sslSniffer/sslSnifferTest/snifftest.c

```diff
@@ -629,45 +709,69 @@ int main(int argc, char** argv)
             }
             else
                 continue;
+
 #ifdef WOLFSSL_SNIFFER_CHAIN_INPUT
-            {
-                unsigned int j = 0;
-                unsigned int remainder = header.caplen;
-
-                chainSz = 0;
-                do {
-                    unsigned int chunkSz;
-
-                    chunkSz = min(remainder, CHAIN_INPUT_CHUNK_SIZE);
-                    chain[chainSz].iov_base = (void*)(packet + j);
-                    chain[chainSz].iov_len = chunkSz;
-                    j += chunkSz;
-                    remainder -= chunkSz;
-                    chainSz++;
-                } while (j < header.caplen);
-            }
+            isChain = 1;
+            j = 0;
+            remainder = header.caplen;
+            chainSz = 0;
+            do {
+                unsigned int chunkSz = min(remainder, CHAIN_INPUT_CHUNK_SIZE);
+                chains[chainSz].iov_base = (void*)(packet + j);
+                chains[chainSz].iov_len = chunkSz;
+                j += chunkSz;
+                remainder -= chunkSz;
+                chainSz++;
+            } while (j < (int)header.caplen);
+            chain = (void*)chains;
+#else
+            chain = (void*)packet;
+            chainSz = header.caplen;
+            (void)isChain;
 #endif
 
-#if defined(WOLFSSL_SNIFFER_CHAIN_INPUT) && \
-    defined(WOLFSSL_SNIFFER_STORE_DATA_CB)
+#ifdef WOLFSSL_ASYNC_CRYPT
+            do {
+                WOLF_EVENT* events[1];
```

Comment:
I don't understand why its array of 1 element of pointer.

Reply:
Same. Part of my todo for setting up a more advanced use case.